compiler/parser/src/lexer.rs

```diff
@@ -1124,7 +1125,7 @@ where
 
     fn eat_single_char(&mut self, ty: Tok) {
         let tok_start = self.get_pos();
-        self.next_char().unwrap();
```

Comment:
is this safe to skip?

Reply:
just removed the `unwrap` call here since it wasn't needed

Reply:
I think this `unwrap` guarantees it never fails. Changing to this one will be zero cost when optimized but check if it is not an error when debugging.

```rust
self.next_char().unwrap_or_else(|| std::hint::unreachable_unchecked());
```

Reply:
this should always be a `Some` since it gets called after a `Some(char)` has been found in `comsume_normal`, I'll add the suggestion though since it definitely makes the intent much clearer.